redash/query_runner/presto.py

```diff
@@ -93,11 +98,17 @@ def get_schema(self, get_stats=False):
         return list(schema.values())
 
     def run_query(self, query, user):
+        enable_user_impersonation = self.configuration.get("user_impersonation")
+        principle_username = None
+        if enable_user_impersonation and user is not None:
+            principle_username = user.email.split('@')[0]
+
         connection = presto.connect(
             host=self.configuration.get("host", ""),
             port=self.configuration.get("port", 8080),
             protocol=self.configuration.get("protocol", "http"),
             username=self.configuration.get("username", "redash"),
+            principle_username=principle_username,
```

Comment:
I just noticed: shouldn't `principle` be named `principal`? The [presto docs](https://docs.starburstdata.com/latest/security/impersonation.html#user-translation-rules) name it `PRINCIPAL`.

I understand [you wrote](https://github.com/dropbox/PyHive/pull/309) the support for this into `pyhive` as well so it might be an easy fix in both places. Might be more trouble than it's worth since changing it now might break other implementations.

Reply:
Yea. This is an autocorrect error that unfortunately no one caught until after it was released in pyhive. (I only just noticed it while I was working on this PR and types in `principal` and it threw an error.)

Making the update in PyHive would be difficult now that it is released, though I can at least implement the correct variable name in the code here.

Reply:
I am reaching out there to work on getting that fix in place. 
https://github.com/dropbox/PyHive/pull/326

Worse case, I'll implement that PR to support both the old and new argument. Lets hold on this PR until a decision is made in PyHive.

Reply:
So they have already merged the correction PR and it will be in the next PyHive release. I'll update this PR with the next release that has that code change in it and update the code to match.

Reply:
I've updated this code to match the changes in the future release of PyHive. This PR will definitely fail checks until that release is out. I will clean up the PR in the `requirements_all_ds.txt` once the new version of pyhive with the change is released.